civihr_employee_portal/src/Mail/OnboardingFormEditNotificationMail.php

```diff
@@ -0,0 +1,157 @@
+
+    $this->formatPersonalDetails($submittedValues);
+    $this->formatPayroll($submittedValues);
+    $this->formatEmergencyContacts($submittedValues);
+    $this->formatDependants($submittedValues);
+
+    // We don't want this info in the email
+    unset($submittedValues['Profile Picture']);
+
+    return $submittedValues;
+  }
+
+  /**
+   * @param array $submittedValues
+   */
+  private function formatPersonalDetails(&$submittedValues) {
+    $personalDetails = &$submittedValues['Personal Details'];
+    unset($personalDetails[0]['Existing Contact']);
+
+    // The options for this are not stored correctly and need manual replacement
+    $ssnApplicationKey = 'I am currently applying for a NI/ SSN';
+    $value = $personalDetails[0][$ssnApplicationKey];
+    $replacement = $value ? 'Yes' : 'No';
+    $personalDetails[0][$ssnApplicationKey] = $replacement;
+  }
+
+  /**
+   * @param $submittedValues
+   */
+  protected function formatPayroll(&$submittedValues) {
+    $payroll = &$submittedValues['Payroll'];
+    $skipStepOption = 'Add Now (Payroll)';
+    if ($this->skippedStep($payroll, $skipStepOption)) {
+      $this->formatSkippedStep($payroll);
+    }
+    unset($payroll[0][$skipStepOption]);
+  }
+
+  /**
+   * @param $submittedValues
+   */
+  private function formatEmergencyContacts(&$submittedValues) {
+    $emergencyContacts = &$submittedValues['Emergency Contact'];
+    $skipStepOption = 'Add Now (Emergency Contacts)';
+
+    if ($this->skippedStep($emergencyContacts, $skipStepOption)) {
+      $this->formatSkippedStep($emergencyContacts);
+      return;
+    }
+    unset($emergencyContacts[0][$skipStepOption]);
+
+    $addSecondOption = 'Add another emergency contact?';
+    $firstContactKey = 'Emergency Contact';
+    $secondContactKey = 'Second Emergency Contact';
+
+    unset($emergencyContacts[$firstContactKey]['Is a dependant?']);
+    unset($emergencyContacts[$secondContactKey]['Is a dependant?']);
+
+    // Check if a second contact was added and remove the value for conditional
+    $addedSecondContact = $emergencyContacts[0][$addSecondOption] === 'Yes';
+    unset($emergencyContacts[0][$addSecondOption]);
+
+    // Remove the second emergency contact if it wasn't added
+    if (!$addedSecondContact) {
+      unset($emergencyContacts[$secondContactKey]);
+    }
+  }
+
+  /**
+   * @param array $submittedValues
+   */
+  private function formatDependants(&$submittedValues) {
+    $dependants = &$submittedValues['Dependants'];
+    $skipStepOption = 'Add Now (Dependants)';
+    if ($this->skippedStep($dependants, $skipStepOption)) {
+      $this->formatSkippedStep($dependants);
+      return;
+    }
+    // Unset unused data
+    unset($dependants[0]);
+    // Remove unused dependant entries
+    $dependantsKeys = ['Fifth', 'Fourth', 'Third', 'Second', 'First'];
+    foreach ($dependantsKeys as $index => $dependantKey) {
+      if (!isset($dependantKey[$index + 1])) {
```

Comment:
The logic here is a bit confusing to me. From the foreach, `$dependantKey` is a string but is being accessed as an array. Or do you mean to write `$dependants`

Reply:
Good catch! It should be `$dependantKeys`.

 I know the logic is a bit complicated here, I tried it in two different ways. Basically I want to go through the list of dependants and unset all those that have not been explicitly added to the form by the user. The user must select "Add another" from the bottom of each dependent input, so I though going through them in reverse order and checking the previous one would work, if you have another suggestion let me know :-)  I'll add a comment to explain it further anyway